features/page-objects/RoomSettingPage.js

```diff
@@ -0,0 +1,34 @@
+const Page = require("./Page");
+
+class RoomSettingPage extends Page {
+  constructor(driver, room) {
+    super(driver);
+    this.room = room;
+  }
+
+  async lock(accessCode) {
+    await this.setAccessLevel('locked');
+    await this.setAccessCode(accessCode);
+    await this.clickUpdateRoom();
+  }
+
+  async setAccessLevel(accessLevel) {
+    const accessLevelSelect = await this.findByCss("[id='room_access_level']");
+    accessLevelSelect.click();
```

Comment:
Not Blocking: Something that may be (but may not be) worth doing is leveraging [JavaScripts `getter` syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) so that we could do something like this:

```javascript
class RoomSettingsPage < Page {
  get accessLevelSelector() {
    return this.findByCss("[id='room_access_level']")
  }

  async setAccessLevel(accessLevel) {
    (await this.accessLevelSelect).click();
    (await this.findByCss(`[value=${accessLevel}]`)).click();
  }
}
```

We could also sprout a "SelectBox" class that takes responsibility for the asyncronicity of clicking/selecting options; but that increases the kinds of test-driver-y objects in the system to Drivers + Page Objects + Component Objects. 

That would look something like this:

```javascript
class SelectTag < Component {
  // this constructor and the component method probably wants to live in Component
  constructor(selector, driver) {
  }
  get component() {
    return this.driver.findByCss(this.selector)
  }
  select(value) {
    return this.component
     .then((el) => el.click())
     .then(this.findByCss(`${selector} [value=${accessLevel}]`))
  }
}
```

I think we'd want to explore this as a group at our next session. Thoughts @KellyAH / @cjoulain / @user512  and @eg-bet?